site/src/pages/tutorials/tutorial-blog-rest-step3.mdx

```diff
@@ -0,0 +1,169 @@
+---
+menuTitle: "Implement CREATE"
+order: 3
+type: step
+tags:
+  - rest
+  - request
+  - request-converter
+level: basic
+---
+
+# Implementing CREATE operation
+
+Now that we've prepared minimal files for the tutorial in [step 2](./tutorial-blog-rest-step2), it's time to start writing the core of the service.
+
+<TutorialSteps current={3} />
+
+In this step, we create a file to contain our service methods. The first service method we write is for creating a blog post. The HTTP method to map with is POST.
+
+By completing this step, you learn how to map your service with HTTP POST (<type://@Post>) and make your own request converter (<type://@RequestConverter>).
+
+## What you need
+
+You need the BlogPost.java file, created from [step 2](./tutorial-blog-rest-step2). You can always [download](https://github.com/line/armeria-examples/blob/master/rest-api-annotated-service/src/main/java/example/armeria/server/blog/BlogPost.java) the full version of this file, instead of creating one yourself.
+
+## 0. Create a service file
+
+1. Create a service file, BlogService.java.
+2. Declare a class, `BlogService`, in the file as shown below. Import required files as you go along the tutorial or copy all from the [sample file](https://github.com/line/armeria-examples/blob/master/rest-api-annotated-service/src/main/java/example/armeria/server/blog/BlogService.java).
+  ```java filename=BlogService.java
+  public final class BlogService {
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final Map<Integer, BlogPost> blogPosts = new ConcurrentHashMap<>();
+    ...
+  ```
+
+## 1. Map HTTP method
+
+Start by [mapping HTTP POST method](/docs/server-annotated-service#mapping-http-service-methods) with our service method, which we will declare at this step:
+
+1. Declare a service method, `createBlogPost()`, in the class `BlogService`.
+2. [Map this service method](/docs/server-annotated-service#mapping-http-service-methods) with the HTTP POST by adding the <type://@Post> annotations.
+3. Assign the endpoint as `/blogs`.
+
+```java filename=BlogService.java highlight=1
+@Post("/blogs")
+public HttpResponse createBlogPost(BlogPost blogPost) throws JsonProcessingException {}
```

Comment:
@minwoox 
I have a question. To keep the consistency with the rest of code samples on the site, 4-space indentations should be used. However, to save up space, I've used 2-space indentations on purpose. Please let me know your preference.

Reply:
We use 4-space indentation in other examples and it's also our convention but I think it's okay to use 2-space indentation because:
- it can save up space as @freevie mentioned 
- there're people who use 2-space and we must not force our convention on them.
@trustin and @ikhoon any thought?